### PR TITLE
feat: HCL output support

### PR DESCRIPTION
**Pull request by robmorgan**
This extends `senc` to natively support HCL output files.

## TODO

Make sure HCL output files are proper HCL. At the moment I think they are still semi-JSON encoded even though some object keys are surrounded in quotes. Example below.

### Current Output

```js
some_attr = {
  "foo" = [
    1,
    2
  ]
  "bar" = true
}
some_block = {
  "some_block_label" = {
    "attr" = "value"
  }
}
```

### Expected Output

```hcl
some_attr = {
    foo = [1, 2]
    bar = true
}

some_block "some_block_label" {
    attr = "value"
}
```

### CODE REVIEW

File: src/engine.rs [hunk 1]
```diff
@@ -692,6 +702,22 @@ mod tests {
...
+            hcl::from_str(&d3.data).expect("error unpacking hcl data");
+        assert_eq!(actual_output3, expected_output);
+
+        let d4: &OutData = &od_vec[3];
+        assert_eq!(d4.out_path, None);
+        assert_eq!(d4.out_ext, Some(String::from(".tf")));
+        assert_eq!(d4.out_prefix, None);
+        let actual_output4: serde_json::Value =
+            hcl::from_str(&d3.data).expect("error unpacking hcl data");
+        assert_eq!(actual_output4, expected_output);
```
Comment (robmorgan) [inline src/engine.rs:720]: I should use a different `expected_output` here so I actually validate the HCL is in the right format vs the tests passing with JSON at the moment.

Comment (yorinasub17) [reply src/engine.rs]: You actually need to validate the string itself, not the serde object. I didn't care much about how the output looks for yaml and json, but obv that matters here with HCL.

File: packages/types/globals.d.ts
Comment (yorinasub17) [inline packages/types/globals.d.ts:1]: Reminder to self: need to release new version of types package. This is currently a manual process...

File: src/engine.rs (hunk 1 above)
Comment (robmorgan) [reply src/engine.rs]: I've switched to comparing `hcl::Value`'s in my latest commit, but I think your point still stands. I need to actually check the string contents.

File: packages/types/globals.d.ts
Comment (robmorgan) [reply packages/types/globals.d.ts:1]: I bumped the version in the types package to `0.0.6` but I forgot to push the commit. Is there anything else I need to do beyond that? Obviously release the package...

Comment (yorinasub17) [reply packages/types/globals.d.ts:1]: Yea i need to run npm publish. Also, the version should really match the gh release. I need to make this part of the release flow but haven't been able to do that yet.

Comment (robmorgan) [reply packages/types/globals.d.ts:1]: Got it!

**Comment by yorinasub17** [comment]
A few more chores before this can be merged:
- every commit needs to be signed off (see https://github.com/apps/dco)
- you need at least one commit message following conventional commits so that the automated release process picks it up.

I recommend squashing all the commits when it's ready for merge and adding the sign off + conventional commits message.

I plan on improving both of these points in the future through squash merges, but need some time to configure all the workflows.

**Comment by robmorgan** [comment]
@yorinasub17 yeah thanks for that. I think I'm going to have to mark this a WIP. Unfortunately until I get the HCL output in an acceptable state, this PR is pretty much going to be useless. I'll keep hacking on it for the time being.

File: packages/types/globals.d.ts
Comment (yorinasub17) [reply packages/types/globals.d.ts:1]: OK I updated the release process so this gets published automatically: https://github.com/fensak-io/senc/pull/58
You also won't need to bump the `package.json` version since that will be handled automatically.

**Comment by robmorgan** [comment]
Got it! I’ll undo my commit

File: src/engine.rs (hunk 1 above)
Comment (robmorgan) [reply src/engine.rs]: Checking string contents now.

**Comment by robmorgan** [comment]
Okay, here's an update.

**TLDR**: It's nontrivial to add an acceptable HCL implementation to `senc`.

I need alot more control over how the resulting HCL is generated, which is very hard when serializing POJOs/JSON objects. I had hoped, as per the [HCL JSON spec](https://github.com/hashicorp/hcl/blob/main/json/spec.md), that a nested JSON block:

```json
"some_block": {
	"some_block_label": {
  	    "attr": "value"
	}
}
```

Would easily resolve to a TF resource definition (labelled block):

```hcl
some_block "some_block_label" {
	attr = "value"
}
```

Unfortunately, HCL2 makes it very hard to do that without using an application-specific schema. The [following blog post](https://log.martinatkins.me/2019/04/25/hcl-json/) provides a good write-up. Pay attention to the "JSON Mapping Ambiguity" section. Additionally, this old [Terraform GitHub issue](https://github.com/hashicorp/hcl/issues/294#issuecomment-1942489568) explains a few more points.

This leaves me in a bit of a pickle. We would need to extend `senc` in two different ways:

1. The higher-level layer would be to maintain a well-formed library of types like CDKTF to help with definitions and appropriately serializing resources ([example](https://github.com/hashicorp/terraform-cdk/blob/e922ce570b48c80d6dedf4d76126b77d2ff92517/packages/cdktf/lib/hcl/render.ts#L202-L239)). I guess that could live outside `senc` and be completely independent.
2. Then the lower level layer would be in the Rust code to combine a Terraform schema somehow if the `OutData` type is `hcl` (or we could possibly introduce a `tf` type). To make matters more complicated, I don't think `hcl-rs` currently supports schemas so we might have to use the [builder pattern](https://github.com/martinohmann/hcl-rs/blob/8349bdd12df96b47d5d7f0bc9ce988dc03c57fb6/crates/hcl-rs/src/ser/mod.rs#L97C1-L117C18) to convert some sort of expected JSON format instead.

So in closing:
1. I can either squash and sign commits and we merge this PR as-is;
2. We leave it open now while we discuss different approaches or;
3. We close it.
